pkg/config/interface.go

```diff
@@ -20,78 +20,78 @@ import (
 	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
 )
 
-// Config defines the interface that project configuration types must follow
+// Config defines the interface that project configuration types must follow.
 type Config interface {
 	/* Version */
 
-	// GetVersion returns the current project version
+	// GetVersion returns the current project version.
 	GetVersion() Version
 
 	/* String fields */
 
-	// GetDomain returns the project domain
+	// GetDomain returns the project domain.
 	GetDomain() string
-	// SetDomain sets the project domain
+	// SetDomain sets the project domain.
 	SetDomain(domain string) error
 
 	// GetRepository returns the project repository.
 	GetRepository() string
-	// SetRepository sets the project repository
+	// SetRepository sets the project repository.
 	SetRepository(repository string) error
 
-	// GetProjectName returns the project name
+	// GetProjectName returns the project name.
 	// This method was introduced in project version 3.
 	GetProjectName() string
-	// SetProjectName sets the project name
+	// SetProjectName sets the project name.
 	// This method was introduced in project version 3.
 	SetProjectName(name string) error
 
-	// GetLayout returns the config layout
+	// GetPluginChain returns the plugin chain.
 	// This method was introduced in project version 3.
-	GetLayout() string
-	// SetLayout sets the Config layout
+	GetPluginChain() []string
```

Comment:
Can we keep this as `GetLayout() []string`? It's nice to have method names that map to YAML fields.

Reply:
I think that the `Config` interface should abstract us from the file, so I think that its methods should be descriptive and I don't think layout fits there. I would change it in the file if it wasn't backwards incompatible.